src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -60,18 +58,64 @@ private Pattern compileAllowlist(List<String> allowlist) {
         return Pattern.compile(joined);
     }
 
+    public String listener() {
+        return listener.toString();
+    }
+
+    static class Listener {
+
+        private static final Pattern PATTERN = Pattern.compile("http://\\[?([0-9a-zA-Z\\-%._:]*)]?:([0-9]+)");
+
+        final String host;
+        final int port;
+
+        public Listener(String host, int port) {
+            this.host = host;
+            this.port = port;
+        }
+
+        public static Listener parseListener(String listener) {
+            Matcher matcher = PATTERN.matcher(listener);
```

Comment:
Why not just lean on java.net.URL() to do the parsing for you?

Reply:
I briefly looked into this and found that it was harder to get what we wanted. It did not seem to support having an empty host and then you had to check that there were no path, query parameters, fragments, etc

Reply:
The regex used is "inspired" from the regex used in Kafka Connect to parse the listener config: https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java#L83